modules/core/src/matmul.dispatch.cpp

```diff
@@ -995,9 +995,18 @@ double Mat::dot(InputArray _mat) const
     CV_INSTRUMENT_REGION();
 
     Mat mat = _mat.getMat();
+    CV_Assert_N( mat.type() == type(), mat.size == size);
```

Comment:
`CV_Assert_N` is deprecated due to malformed or useless error messages.
Use `CV_Check*` instead.